### ISSUE TITLE
if still functional, may you provide a minimal working example?

### ISSUE THREAD

**Issue by nbehrnd** [samwilson/commons-category-downloader#1]
Your tool is mentioned among the wikimedia [download tools](https://commons.wikimedia.org/wiki/Commons:Download_tools) as working from the CLI.

Aiming for a list of url about illustrations about a category and subsequent download by wget2, a task once (March 2021) working well with [wikilovesdownloads](https://wikilovesdownloads.toolforge.org/) yet not now.  In case _your_ application retained its functionality, may you please complement the documentation to the effect

+ dependencies of your program to resolve vs. other libraries/modules are listed
+ one/two typical small cases of application are presented to get familiar with your application's working

The intended OS to work with your application would be Linux Debian 12/bookworm, branch testing.

**Comment by TwitchyMcJoe** [comment]
Just got it working on and off this morning.

What I needed to do on Windows:

0: download these files from Git and extract them somewhere
1: Install php (You will need to add php.exe to your environment path)
2: Install composer (https://getcomposer.org/download/)
3: run composer command line (composer install) in the directory you extracted in step 0
4: update dependencies (composer update) also in the same directory as step 0
5: navigate to Client.php in the new vendor\guzzlehttp\guzzle\src folder that appeared in your step 0 folder
6: change the defaults (starting in line 231) to
 ` $defaults = [
            'allow_redirects' => RedirectMiddleware::$defaultSettings,
            'http_errors' => true,
            'decode_content' => true,
            'verify' => false,
            'cookies' => true,
            'idn_conversion' => false,`
7: in your main folder for this, add this to download.php on line 13 (that's where I thought made sense)
`ini_set( 'user_agent', 'CommonsCategoryDownloader/0.0 (https://github.com/samwilson/commons-category-downloader; coolbot@example.org)' );
`
8: change the category you want to download ($fspsCat) in download.php, 

9: create a folder called "files" in the directory you made in step 0 (or change $localFile to a location of your choice)

10: run php download.php from command prompt.

**Comment by nbehrnd** [comment]
Thank yo for the note.  Meanwhile (with a transition to Debian 13/trixie,
branch testing), I returned multiple times to the special web site to generate
url eventually accessed successfully again.  This includes smaller as well as
larger sets of data (e.g., as input for `dek_fetch_1.py` to assemble an Anki
deck for one form of shorthand, https://github.com/nbehrnd/DEK_VS_svg, approx.
39k of files).

**Comment by samwilson** [comment]
Yes sorry about the lack of documentation! This script hasn't seen much development.

I'm working to incorporate its functionality into [mwcli](https://github.com/samwilson/mwcli).

If you're looking to download all of a [user's contributions](https://github.com/samwilson/mwcli#exportcontribs), then `mwcli export:contribs --user=ExampleUser` works for that. A single-category-tree variant is coming soon I hope.
